### PR TITLE
NTBS-2785: SQL for Cohort Review

### PR DESCRIPTION
**Pull request by LeoCie**
I've run all new queries on the devbox, and most don't take longer than a couple of seconds. The only query that we may need to worry about is the manual test results query for ETS case data, this takes ~20-40s but it is only run once so it may not be a huge issue.

### CODE REVIEW

**Comment by LeoCie** [comment]
@aptemus that does help, but it doesn't give the time period. Thought it might be useful to say "Yes - Current" or "Yes - in last 5 years". It isn't a lot of logic to add a string to the end of another, but wondered if we'd want to do it here

File: source/dbo/Stored Procedures/Power BI Reporting/uspGenerateNtbsCaseRecord.sql [hunk 1]
```diff
@@ -15,6 +29,25 @@ BEGIN TRY
 		WHERE rr.SourceSystem = 'NTBS'
 		GROUP BY rr.NotificationId;
 
+	INSERT INTO @TempManualTestResult
+	SELECT DISTINCT rr.NotificationId, ManualTestTypeId,
```
Comment (aptemus) [inline source/dbo/Stored Procedures/Power BI Reporting/uspGenerateNtbsCaseRecord.sql:33]: You could use a where clause to exclude Chest x-ray (and the upcoming Chest CT) test types here, as you won't be trying to summarise them for this anyway?

File: source/dbo/Stored Procedures/Power BI Reporting/uspGenerateReportingRecords.sql [hunk 2]
```diff
@@ -100,6 +101,7 @@ BEGIN TRY
 		,p.NhsNumber										AS NhsNumberToLookup --this will later be reformatted if valid
 		,p.GivenName										AS Forename
 		,p.FamilyName										AS Surname
+		,LEFT(p.GivenName,1)+'.'+LEFT(p.FamilyName,1)+'.'   AS Initials
```
Comment (aptemus) [inline source/dbo/Stored Procedures/Power BI Reporting/uspGenerateReportingRecords.sql:104]: Not sure I would have gone as far as storing the initials in SQL server but as you have, I would just store the initials with no dots between them.

File: source/dbo/Stored Procedures/Power BI Reporting/uspGenerateReportingRecords.sql [hunk 3]
```diff
@@ -125,6 +128,7 @@ BEGIN TRY
 		,p.NhsNumber										AS NhsNumberToLookup --this will later be reformatted if valid
 		,p.Forename											AS GivenName
 		,p.Surname											AS FamilyName
+		,LEFT(p.Forename,1)+'.'+LEFT(p.Surname,1)+'.'		AS Initials
```
Comment (aptemus) [inline source/dbo/Stored Procedures/Power BI Reporting/uspGenerateReportingRecords.sql:131]: Bit hard to follow why this is in twice but this type of transform that can be done on all records can just be done once at the end of the case data SP, in uspGenerateReportingCaseData

File: source/dbo/Tables/Power BI Reporting/Record_CaseData.sql [hunk 4]
```diff
@@ -52,12 +52,15 @@
 	[MdrTreatmentDate] date NULL,
 	[EnhancedCaseManagement] [varchar] (10) NULL,
 	[EnhancedCaseManagementLevel] [nvarchar] (10) NULL,
+	[FirstPresentationSetting] [nvarchar] (25) NULL,
```
Comment (aptemus) [inline source/dbo/Tables/Power BI Reporting/Record_CaseData.sql:55]: I don't think this will be long enough - the free text healthcare setting description field is 100 characters in the database.

File: source/dbo/Stored Procedures/Power BI Reporting/uspGenerateNtbsCaseRecord.sql (hunk 1 above)
Comment (LeoCie) [reply source/dbo/Stored Procedures/Power BI Reporting/uspGenerateNtbsCaseRecord.sql:33]: Yep added where clause

File: source/dbo/Stored Procedures/Power BI Reporting/uspGenerateEtsCaseRecord.sql [hunk 5]
```diff
@@ -18,6 +42,55 @@ BEGIN TRY
...
+				WHEN 3 THEN 'Awaiting'
+				ELSE 'No result'
+			END) OVER (PARTITION BY rr.NotificationId, mttm.NtbsId ORDER BY
+			CASE lr.Result
+				WHEN 0 THEN 2
+				WHEN 1 THEN 1
+				WHEN 2 THEN 3
+				WHEN 3 THEN 3
+				ELSE 4
+			END) AS [Result]
```
Comment (LeoCie) [inline source/dbo/Stored Procedures/Power BI Reporting/uspGenerateEtsCaseRecord.sql:60]: @benjimarshall A nicer way of doing this? Or is this the best we're gonna get?
